security/pkg/platform/onprem.go

```diff
@@ -84,6 +89,11 @@ func (ci *OnPremClientImpl) GetServiceIdentity() (string, error) {
 		return "", err
 	}
 	if len(serviceIDs) != 1 {
+		for _, s := range serviceIDs {
+			if strings.HasPrefix(s, "spiffe://") {
+				return serviceIDs[0], nil
```

Comment:
Return s instead of serviceIDs[0]?

Reply:
This function gets the identity from the service (on VM) cert, which should be single value. Do we want to change this?

Reply:
It is not single value for pilot and other components that need to interoperate with non-spiffe clients.

Reply:
Oh I see. You are right.